src/Controls/src/Core/AnimationExtensions.cs

```diff
@@ -260,6 +273,14 @@ static void AnimateInternal<T>(IAnimatable self, IAnimationManager animationMana
 
 			s_animations[key] = info;
 
+			if (self is VisualElement ve)
+			{
+				ve.Unloaded += (s, e) =>
+				{
+					s_animations.Remove(key);
+				};
```

Comment:
Can the method used to subscribe to `Unloaded` be `static`? This currently creates a closure on `key`, might leak the `key` object? Otherwise, it feels like something needs to unsubscribe from this event.

Reply:
From unit tests this will not leak. The `Unloaded` event has the same lifetime as the object, but I can change it to be static to avoid future leaks... But how I'll access the `key` if the handler is static?

Reply:
The test doesn't verify the `AnimationKey` doesn't leak and might not be able to. It seems like it would be kept alive as long as the `VisualElement`.

Maybe we could unsubscribe when the event fires, and that is a start?

Reply:
@jonathanpeppers  Done, let me know if it looks good for you (: